src/hugolib/site_render.go

```diff
@@ -357,7 +357,11 @@ func (s *Site) renderAASA() error {
 		return errors.New("failed to create targetPath for Apple App Site Associate")
 	}
 
-	templ := s.lookupLayouts("apple_app_site_association.json", "_default/apple_app_site_association.json", "_internal/_default/apple_app_site_association.json")
+	templ := s.lookupLayouts("apple_app_site_association_v2.json", "_default/apple_app_site_association_v2.json", "_internal/_default/apple_app_site_association_v2.json")
+
+	if s.Cfg.GetString("aasaVersion") == "v1" {
```

Comment:
This if block here needs to cover both cases. Right now you're loading the v2 template even when the user is requesting v1.

Reply:
Thanks for the feedback. I updated the logic to only set the v2 template if the aasaVersion is not set to 1